zanata-war/src/main/java/org/zanata/webtrans/server/HibernateIntegrator.java

```diff
@@ -37,6 +38,8 @@ public void integrate(Configuration configuration,
                     updateListener);
             eventListenerRegistry.appendListeners(EventType.POST_INSERT,
                     updateListener);
+            eventListenerRegistry.appendListeners(EventType.POST_COMMIT_UPDATE,
+                    ServiceLocator.instance().getInstance(SlugEntityUpdatedListener.class));
```

Comment:
That is correct. The POJO would use the service locator to get a stateless bean and execute it's logic.